bigbluebutton-html5/imports/ui/components/tabs/components/TabContent/TabContentContainer.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { withTracker } from 'meteor/react-meteor-data';
+import Service from '../../service';
+import PresentationService from '/imports/ui/components/presentation/presentation-uploader/service';
+import TabContent from './TabContent';
+
+const TabContentContainer = (props) => {
+  const options = [];
+
+  const allOptionPages = [];
+
+  let Pages = null;
+
+  const {
+    presentations, fileType, onSelectoption, selectedOption,
+  } = props;
+
+  presentations.forEach((element) => {
```

Comment:
as discussed do not fetch all the presentation, fetch only `id, filename` to fill the dropdown, create function `getPresentationDropdownValues` and pass data as we require for dropdown `[{ label, value }]`, do not loop at component level

Reply:
fixed.